### ISSUE TITLE
CacheOnReadFs.Stat() returns FileNotFound when file exists on base but not on cache

### ISSUE THREAD

**Issue by ns-cweber** [spf13/afero#115]
I'm on OS X 10.11.6

Given directories `/tmp/base/` and `/tmp/cache` wherein the former contains a file `foo` and the latter is empty, the following code fails with the error message: `CACHE_ON_READ: stat /tmp/cache/foo: no such file or directory`. Because `foo` exists in `base`, I was expecting to see a valid FileInfo and no error.

```go
func main() {
	base := afero.NewBasePathFs(afero.NewOsFs(), "/tmp/base/")
	if _, err := base.Stat("foo"); err != nil {
		log.Fatalln("BASE:", err)
	}
	fs := afero.NewCacheOnReadFs(
		base,
		afero.NewBasePathFs(afero.NewOsFs(), "/tmp/cache/"),
		time.Duration(0),
	)
	if _, err := fs.Stat("foo"); err != nil {
		log.Fatalln("CACHE_ON_READ:", err)
	}
}
```

**Comment by uppfinnarn** [comment]
Trivially reproducible with `MemMapFs` instances. This is a slight modification of the test case in the README:

```go
func TestCacheOnReadFs(t *testing.T) {
	appFS := afero.NewOsFs()
	// create test files and directories
	appFS.MkdirAll("src/a", 0755)
	afero.WriteFile(appFS, "src/a/b", []byte("file b"), 0644)
	afero.WriteFile(appFS, "src/c", []byte("file c"), 0644)
	name := "src/c"
	cacheFS := afero.NewCacheOnReadFs(appFS, afero.NewMemMapFs(), 1*time.Hour)
	_, err := cacheFS.Stat(name)
	if os.IsNotExist(err) {
		t.Errorf("file \"%s\" does not exist.\n", name)
	}
}
```

**Comment by mbertschler** [comment]
Fixed via #135
